olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,142 @@
+  void SetDataSize(int64_t value) { data_size_ = value; }
+
+  const Data& GetData() const { return data_; }
+  void SetData(Data value) { data_ = std::move(value); }
+
+  const boost::optional<std::string>& GetDataHandle() const {
+    return data_handle_;
+  }
+  void SetDataHandle(std::string value) { data_handle_ = std::move(value); }
+
+  const boost::optional<int64_t>& GetTimestamp() const { return timestamp_; }
+  void SetTimestamp(int64_t value) { timestamp_ = value; }
+
+ private:
+  /// A key that specifies which
+  /// [Partition](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html)
+  /// the content is related to. This is provided by the user while producing to
+  /// the stream layer. The maximum length of the partition key is 500
+  /// characters.
+  std::string partition_;
+
+  /// The checksum of the content. The algorithm used to calculate the checksum
+  /// is user specific. Algorithms that can be used are, for example, MD5 or
+  /// SHA1. This is not a secure hash, it's used only to detect changes in
+  /// content.
+  boost::optional<std::string> checksum_;
+
+  /// The compressed size of the content in bytes. Applicable also when
+  /// `Content-Encoding` is set to gzip when uploading and downloading data.
+  /// Note: This will be present only if �_dataHandle_� field is present.
+  boost::optional<int64_t> compressed_data_size_;
+
+  /// The nominal size in bytes of the content. When compression is enabled,
+  /// this is the size of the uncompressed content. Note: This will be present
+  /// only if the �dataHandle� field is present.
+  boost::optional<int64_t> data_size_;
+
+  /// The content published directly in the metadata and encoded in base64. The
+  /// size of the content is limited. Either `data` or `dataHandle` must be
+  /// present. Data Handle Example: 1b2ca68f-d4a0-4379-8120-cd025640510c. Note:
+  /// This will be present only if the message size is less than or equal to 1
+  /// MB.
+  Data data_;
+
+  /// The handle created when uploading the content. It is used to retrieve the
+  /// content at a later stage. Either `data` or `dataHandle` must be present.
+  /// Note: This will be present only if the message size is greater than 1 MB.
+  boost::optional<std::string> data_handle_;
+
+  /// The timestamp of the content, in milliseconds since the Unix epoch. This
+  /// can be provided by the user while producing to the stream layer. Refer to
+  /// [NewPartition
+  /// Object](https://developer.here.com/olp/documentation/data-client-library/api_reference_scala/index.html#com.here.platform.data.client.scaladsl.NewPartition).
+  /// If not provided by the user, this is the timestamp when the message was
+  /// produced to the stream layer.
+  boost::optional<int64_t> timestamp_;
+};
+
+/**
+ * @brief Represents a message read from a stream layer.
+ */
+class DATASERVICE_READ_API Message final {
```

Comment:
Add doxygen to each method please. It is a public class after all.
Same for below Messages.